### PR TITLE
internal/gocdk: add code for prompting the user and optionally saving the response

### PR DESCRIPTION
**Pull request by vangent**

### CODE REVIEW

File: internal/cmd/gocdk/internal/prompt/prompt.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+// TODO(rvangent): Add support for validation?
+// TODO(rvangent): Ctrl-C doesn't work; maybe replace cancel.
+func String(in io.Reader, out io.Writer, msg, dflt string) (string, error) {
+	fmt.Fprintln(out)
+	fmt.Fprintln(out, msg)
+	if dflt != "" {
+		fmt.Fprintln(out, "  default:", dflt)
+	}
+	const prompt = `Enter a value, or "cancel": `
+	reader := bufio.NewReader(in)
```
Comment (zombiezen) [inline internal/cmd/gocdk/internal/prompt/prompt.go]: This can be tricky: if `in` is a `*bufio.Reader`, this function will simply pass it through, but otherwise this can read more than just the particular line and then discard the data. This can result in unexpected behavior if `String` is called repeatedly.

Probably best to take in a `*bufio.Reader` directly and push responsibility to the caller for arranging for the correct thing to happen.

File: internal/cmd/gocdk/internal/prompt/prompt.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	"io"
+	"strings"
+
+	"gocloud.dev/internal/cmd/gocdk/internal/static"
+)
+
+// String prompts the user to input a string. It outputs msg to out, then
+// prompts the user to enter a value. If the user just hits return, the
+// value defaults to dflt.
+// The function only returns an error if the user enters "cancel".
```
Comment (zombiezen) [inline internal/cmd/gocdk/internal/prompt/prompt.go]: And on I/O errors.

File: internal/cmd/gocdk/internal/prompt/prompt.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	}
+	const prompt = `Enter a value, or "cancel": `
+	reader := bufio.NewReader(in)
+	for {
+		fmt.Fprintf(out, prompt)
+		s, err := reader.ReadString('\n')
+		if err != nil {
+			return "", err
+		}
+		s = strings.Replace(s, "\n", "", -1)
```
Comment (zombiezen) [inline internal/cmd/gocdk/internal/prompt/prompt.go]:
```suggestion
    s = strings.TrimSuffix(s, "\n")
```

File: internal/cmd/gocdk/internal/prompt/prompt.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+	"gocloud.dev/internal/cmd/gocdk/internal/static"
+)
+
+// String prompts the user to input a string. It outputs msg to out, then
+// prompts the user to enter a value. If the user just hits return, the
+// value defaults to dflt.
+// The function only returns an error if the user enters "cancel".
+// TODO(rvangent): Add support for validation?
+// TODO(rvangent): Ctrl-C doesn't work; maybe replace cancel.
```
Comment (zombiezen) [inline internal/cmd/gocdk/internal/prompt/prompt.go]: Ctrl-C will work if you accept a `context.Context`.

File: internal/cmd/gocdk/internal/prompt/prompt.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+func String(in io.Reader, out io.Writer, msg, dflt string) (string, error) {
+	fmt.Fprintln(out)
```
Comment (zombiezen) [inline internal/cmd/gocdk/internal/prompt/prompt.go]: This prompt seems verbose to me. IIUC the result is:

```
Enter a GCP region. default: us-central1
Enter a value, or "cancel":
```

What about:

```
Enter a GCP region [default: us-central1], or "cancel": 
```

(We could remove the `or "cancel"` if we support Ctrl-C, see above.)

File: internal/cmd/gocdk/internal/static/static.go [hunk 6]
```diff
@@ -132,8 +133,27 @@ func AddProvider(provider string) *Action {
 	}
 }
 
+// ReadLocal reads a local Terraform variable from main.tf.
+// The local variable was likely added using an Action returned from AddLocal.
+// If the variable is not found, it returns the empty string and no error.
+func ReadLocal(biomeDir, key string) (string, error) {
```
Comment (zombiezen) [inline internal/cmd/gocdk/internal/static/static.go]: Organization: This function doesn't feel like it conceptually belongs under the heading "static files" to me. Perhaps we should change the package name to reflect, but probably doesn't belong in this PR.

File: internal/cmd/gocdk/internal/prompt/prompt.go (hunk 1 above)
Comment (vangent) [reply internal/cmd/gocdk/internal/prompt/prompt.go]: Done.

File: internal/cmd/gocdk/internal/prompt/prompt.go (hunk 2 above)
Comment (vangent) [reply internal/cmd/gocdk/internal/prompt/prompt.go]: Done.

File: internal/cmd/gocdk/internal/prompt/prompt.go (hunk 3 above)
Comment (vangent) [reply internal/cmd/gocdk/internal/prompt/prompt.go]: Done.

File: internal/cmd/gocdk/internal/prompt/prompt.go (hunk 4 above)
Comment (vangent) [reply internal/cmd/gocdk/internal/prompt/prompt.go]: Updated TODO.

File: internal/cmd/gocdk/internal/prompt/prompt.go (hunk 5 above)
Comment (vangent) [reply internal/cmd/gocdk/internal/prompt/prompt.go]: I was trying to follow what Terraform does. We can revisit how we want prompts to look later; added a TODO.

File: internal/cmd/gocdk/internal/static/static.go (hunk 6 above)
Comment (vangent) [reply internal/cmd/gocdk/internal/static/static.go]: Agreed; added a TODO.

**Comment by vangent** [comment]
Thanks for the quick review; PTAL.
